pandas/core/construction.py

```diff
@@ -574,6 +574,25 @@ def sanitize_array(
     return subarr
 
 
+def range_to_ndarray(rng: range) -> np.ndarray:
+    """
+    Cast a range object to ndarray.
+    """
+    # GH#30171 perf avoid realizing range as a list in np.array
+    try:
+        arr = np.arange(rng.start, rng.stop, rng.step, dtype="int64")
```

Comment:
isn't this very simliar to maybe_cast_to_integer_array and generate_regular_range

Reply:
maybe_cast_to_integer_array no, thats the crux of #40110 

generate_regular_range conceptually, but they take different args so ill have to take a closer look

Reply:
ok as long as its on the refactor / restructure list